graphviz/src/main/java/io/deephaven/graphviz/GraphVizBuilder.java

```diff
@@ -0,0 +1,74 @@
+package io.deephaven.graphviz;
+
+import guru.nidi.graphviz.attribute.Label;
+import guru.nidi.graphviz.attribute.Shape;
+import guru.nidi.graphviz.model.MutableGraph;
+import guru.nidi.graphviz.model.MutableNode;
+import io.deephaven.qst.table.LabeledTable;
+import io.deephaven.qst.table.LabeledTables;
+import io.deephaven.qst.table.ParentsVisitor;
+import io.deephaven.qst.table.TableSpec;
+
+import java.util.LinkedHashMap;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import static guru.nidi.graphviz.model.Factory.mutGraph;
+import static guru.nidi.graphviz.model.Factory.mutNode;
+import static guru.nidi.graphviz.model.Factory.to;
+
+class GraphVizBuilder {
+
+    static MutableGraph of(Iterable<TableSpec> tables) {
+        NodesBuilder consumer = new NodesBuilder();
+        ParentsVisitor.postOrderWalk(tables, consumer);
+        MutableGraph graph = mutGraph().setDirected(true);
+        for (MutableNode node : consumer.identifiers.values()) {
+            graph.add(node);
+        }
+        AppendLinks.ofAll(consumer.identifiers);
+        return graph;
+    }
+
+    static MutableGraph of(LabeledTables tables) {
+        NodesBuilder consumer = new NodesBuilder();
+        ParentsVisitor.postOrderWalk(tables.tables(), consumer);
+        MutableGraph graph = mutGraph().setDirected(true);
+        for (MutableNode node : consumer.identifiers.values()) {
+            graph.add(node);
+        }
+        AppendLinks.ofAll(consumer.identifiers);
+
+        int argIndex = 0;
+        for (LabeledTable e : tables) {
+            String label = e.label();
+            TableSpec table = e.table();
+
+            MutableNode argNode = mutNode(Label.of(label)).add(Shape.NOTE);
+            graph.add(argNode);
+
+            MutableNode other = Objects.requireNonNull(consumer.identifiers.get(table));
+            argNode.addLink(to(other).with(Label.of(Integer.toString(argIndex))));
```

Comment:
what purpose does this serve in the final output? there are only two tests, and only one has the labels turned on such that we use this feature (...i'm only 80% that this is indeed the line which causes the label in question), but three edges in the graph just get a "0", "1", "2" label on them, despite the nodes already being labeled. The ambiguous edges are already labeled too, "left" and "right".

Reply:
Good point, the inputs are already labeled.